ecosystem/typescript/sdk_v2/tests/unit/deserializer.test.ts

```diff
@@ -132,13 +132,15 @@ describe("BCS Deserializer", () => {
 
   it("deserializes a single deserializable class", () => {
     // Define the MoveStruct class that implements the Deserializable interface
-    class MoveStruct implements Serializable {
+    class MoveStruct extends Serializable {
```

Comment:
the whole point with deserializable and serializable is to later create a Move type class (MoveU8, MoveU16, etc) and have each of them to be deserializable and serializable? correct? I wonder how we can achieve it with this structure as in typescript a class can only extends one class.

Reply:
The point of the Serializable class is partially to encourage developers to use a more intuitive structure for BCS-serializable data and also to allow for composable serialization. Plus the `bcsToBytes` function

The deserialization aspect of things is mostly a separate process in the sense that it doesn't require an instance of a Deserializable class type to deserialize a byte buffer, since it just uses the static deserialize function. That's why it can be an internal interface, because it doesn't actually do anything. The point of `Deserializable` is to be an interface to ensure that when you call `deserializer.deserialize(something)` something is actually deserializable

But we don't necessarily need Serializable to be a class, I suppose, if you think that will lead to issues down the road. It could just remain an interface

Reply:
@0xmaayan I'm going to keep it simple and just leave `Serializable` as a class for now, but to be clear, `Deserializable` is an internal interface that exists solely to facilitate the more intuitive syntax with `Deserializer` instances; that is:

```typescript
const myClassInstance = deserializer.deserialize(MyClass);
// as opposed to
const myClassInstance = MyClass.deserialize(deserializer);
```